### PR TITLE
Task/m fc imaged nodes

### PR DESCRIPTION
**Pull request by abhimutant**
DataSource for Imaged Node Details

### CODE REVIEW

File: nutanix/resource_nutanix_foundation_central_image_nodes.go [hunk 1]
```diff
@@ -313,14 +655,24 @@ func expandCommonNetworkSettings(d *schema.ResourceData) *fc.CommonNetworkSettin
...
+	cns.CvmDNSServers = expandNetworklist(settingsMap["cvm_dns_servers"])
+	cns.CvmNtpServers = expandNetworklist(settingsMap["cvm_ntp_servers"])
+	cns.HypervisorDNSServers = expandNetworklist(settingsMap["hypervisor_dns_servers"])
+	cns.HypervisorNtpServers = expandNetworklist(settingsMap["hypervisor_ntp_servers"])
 
 	return &cns
 }
 
+func expandNetworklist(pr interface{}) []string {
+	c := make([]string, 0)
```
Comment (bhati-pradeep) [inline nutanix/resource_nutanix_foundation_central_image_nodes.go]:
```suggestion
        prList := pr.([]interface{})
	c := make([]string, len(prList))
```

File: nutanix/resource_nutanix_foundation_central_image_nodes.go [hunk 2]
```diff
@@ -313,14 +655,24 @@ func expandCommonNetworkSettings(d *schema.ResourceData) *fc.CommonNetworkSettin
...
+	prList := pr.([]interface{})
+
+	for _, k := range prList {
```
Comment (bhati-pradeep) [inline nutanix/resource_nutanix_foundation_central_image_nodes.go]:
```suggestion
	for k, v := range prList {
```

File: nutanix/resource_nutanix_foundation_central_image_nodes.go [hunk 3]
```diff
@@ -313,14 +655,24 @@ func expandCommonNetworkSettings(d *schema.ResourceData) *fc.CommonNetworkSettin
...
+		c = append(c, k.(string))
```
Comment (bhati-pradeep) [inline nutanix/resource_nutanix_foundation_central_image_nodes.go:671]: c[k] = v.(string)

Comment (bhati-pradeep) [inline nutanix/resource_nutanix_foundation_central_image_nodes.go]:
```suggestion
		c[k] = v.(string)
		
```

File: nutanix/resource_nutanix_foundation_central_image_nodes.go [hunk 4]
```diff
@@ -466,10 +826,35 @@ func resourceNutanixFCImageClusterCreate(ctx context.Context, d *schema.Resource
 	}
 	req.Timezone = utils.StringPtr(timezone.(string))
 
+	skipClusterCreation, ok := d.GetOk("skip_cluster_creation")
+	if !ok {
+		req.SkipClusterCreation = false
+	}
+	req.SkipClusterCreation = skipClusterCreation.(bool)
```
Comment (bhati-pradeep) [inline nutanix/resource_nutanix_foundation_central_image_nodes.go]: This is overriding the changes done by if condition at line 830. By default req.SkipClusterCreation would be false when the struct is initialized.
```suggestion
	if skipClusterCreation, ok := d.GetOk("skip_cluster_creation"); ok {
	        req.SkipClusterCreation = skipClusterCreation.(bool)
	}
```

Comment (bhati-pradeep) [reply nutanix/resource_nutanix_foundation_central_image_nodes.go:833]: suggestion for line 829-833

File: nutanix/resource_nutanix_foundation_central_image_nodes.go [hunk 5]
```diff
@@ -497,17 +882,80 @@ func resourceNutanixFCImageClusterCreate(ctx context.Context, d *schema.Resource
...
+	// append errors for failed node imaging
+	for _, v := range progress.ClusterStatus.NodeProgressDetails {
+		if utils.Float64Value(v.PercentComplete) < float64(aggregatePercentComplete) {
+			message := ""
+			for _, v1 := range v.MessageList {
+				message += *v1
+			}
+			diags = append(diags, diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  fmt.Sprintf("Node imaging for imaged_node_uuid IP: %s failed with error:  %s.", *v.ImagedNodeUUID, *v.Status),
```
Comment (bhati-pradeep) [inline nutanix/resource_nutanix_foundation_central_image_nodes.go:922]: `imaged_node_uuid IP` can we have something else as an identifier like cvm IP or something using which user can easily identify the node.

**Review by bhati-pradeep — APPROVED** [review]

**Review by siddharth-kulshrestha — APPROVED** [review]

**Pull request merged**